x/poolmanager/taker_fee_test.go

```diff
@@ -0,0 +1,90 @@
+package poolmanager_test
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/osmomath"
+	"github.com/osmosis-labs/osmosis/v19/app/apptesting"
+)
+
+// validates that the pool manager keeper can charge taker fees correctly.
+// If the sender is whitelisted, then the taker fee is not charged.
+// Otherwise, the taker fee is charged.
+func (s *KeeperTestSuite) TestChargeTakerFee() {
+
+	var (
+		defaultTakerFee = osmomath.MustNewDecFromStr("0.01")
+		defaultAmount   = sdk.NewInt(100)
+	)
+
+	tests := map[string]struct {
+		shouldSetSenderWhitelist bool
+		tokenIn                  sdk.Coin
+		tokenOutDenom            string
+		sender                   sdk.AccAddress
+		exactIn                  bool
+		takerFee                 osmomath.Dec
+
+		expectedResult sdk.Coin
+		expectError    error
+	}{
+		"fee charged on token in": {
+			takerFee:      defaultTakerFee,
+			tokenIn:       sdk.NewCoin(apptesting.ETH, defaultAmount),
+			tokenOutDenom: apptesting.USDC,
+			sender:        s.TestAccs[0],
+			exactIn:       true,
+
+			expectedResult: sdk.NewCoin(apptesting.ETH, defaultAmount.ToLegacyDec().Mul(osmomath.OneDec().Sub(defaultTakerFee)).TruncateInt()),
+		},
+		"fee bypassed due to sender being whitelisted": {
+			takerFee:                 defaultTakerFee,
+			tokenIn:                  sdk.NewCoin(apptesting.ETH, defaultAmount),
+			tokenOutDenom:            apptesting.USDC,
+			sender:                   s.TestAccs[0],
+			exactIn:                  true,
+			shouldSetSenderWhitelist: true,
+
+			expectedResult: sdk.NewCoin(apptesting.ETH, defaultAmount),
+		},
+		// TODO: under more test cases
+		// https://github.com/osmosis-labs/osmosis/issues/6633
+		// - exactOut: false
+		// - sender does not have enough coins
```

Comment:
another test that should be added (or added to the todo), is that the whitelist is set but the address is not the one that is whitelisted, should still be charged the fee. I think it would be a simple add and would be a nice thing to confirm.

Reply:
Added